### ISSUE TITLE
tui_flush: Assertion `r.bot < grid->height && r.right < grid->width' failed. Core dumped

### ISSUE THREAD

**Issue by sanga** [neovim/neovim#8774]
- `nvim --version`:

```
NVIM v0.3.2-24-gc8e7a447c
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/gcc-5 -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector-strong -fdiagnostics-color=auto -Wno-array-bounds -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-69e7482a-bb69-4322-8cfc-509faa4952d2
Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"
   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/travis/build/neovim/bot-ci/build/neovim/build/nvim.AppDir/usr/share/nvim
"
Run :checkhealth for more info

```- Vim (version: ) behaves differently?
- Operating system/version: linux (ubuntu 18.04)
- Terminal name/version: alacritty (current head of "scrollback" branch)
- `$TERM`: xterm-256-color

### Steps to reproduce using `nvim -u NORC`

```
steps to repro this are basically:
- run vim
- run this "!echo -n "tim\nwas\nhere" && sleep 10"
- resize the terminal window whilst the sleep is ongoing

Basically, I guess the point is to have mulitple lines there and resize

```

### Actual behaviour

### Expected behaviour
No failed assert

**Comment by KillTheMule** [comment]
> NVIM v0.2.3-1-gd6f9d1df0

As a first step, you should update to neovim 0.3.1,it's been released some days ago, and in particular of course includes all the fixes that went into 0.3 over .2.2, which is quite a lot.

**Comment by sanga** [comment]
apologies. I just blindly executed that command without thinking. The version is actually: `v0.3.2-24-gc8e7a447c`. Updating bug description

**Comment by KillTheMule** [comment]
I'm on v0.3.0-1353-gb33d304, using st 0.7, and cannot reproduce this. Also tried with xterm, no problem. Could you give another terminal emulator a shot?

**Comment by sanga** [comment]
xterm gives me the same problem. I'm on i3 if that makes any difference

**Comment by KillTheMule** [comment]
Ok, I think the devs need to come in :) Your `$TERM` looks funky, though, could you try `xterm-256color`? I'm not sure about alacritty (I'm not even sure this is important), but for xterm you could give `TERM=xterm` a shot.

Just to make sure: You ARE running `nvim -u NORC`?

**Comment by bfredl** [comment]
might be related to #8221, I'm currently fixing a very similar bug on master.

**Comment by sanga** [comment]
still present in NVIM v0.3.2-30-g72ddbb675 (I thought https://github.com/neovim/neovim/pull/8771 may help)

**Comment by bfredl** [comment]
Good to know, I will look into it!

**Comment by kedashoe** [comment]
Getting this same error. Also happening with any window resize, or if I open a 2nd tab in my terminal (using simple built in ubuntu terminal).

```bash
$ nvim --version
NVIM v0.3.2-dev
Build type: RelWithDebInfo
Lua 5.1
Compilation: /usr/bin/x86_64-linux-gnu-gcc -g -O2 -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wvla -fstack-protector --param ssp-buffer-size=4 -Wno-array-bounds -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim-ivk5_s/neovim-0.3.0~ubuntu1+git201807252033-9c0c5e8-479a1d0/build/config -I/build/neovim-ivk5_s/neovim-0.3.0~ubuntu1+git201807252033-9c0c5e8-479a1d0/src -I/build/neovim-ivk5_s/neovim-0.3.0~ubuntu1+git201807252033-9c0c5e8-479a1d0/.deps/usr/include -I/usr/include -I/build/neovim-ivk5_s/neovim-0.3.0~ubuntu1+git201807252033-9c0c5e8-479a1d0/build/src/nvim/auto -I/build/neovim-ivk5_s/neovim-0.3.0~ubuntu1+git201807252033-9c0c5e8-479a1d0/build/include
Compiled by buildd@lgw01-amd64-038

Features: +acl +iconv +jemalloc +tui 
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
Thanks!

**Comment by kedashoe** [comment]
Got a release including https://github.com/neovim/neovim/commit/5f15788dc3ac1ea7906673eacc9cf9bdb1f14212, everything seems to be working now. Since I started using nvim a year or so ago you all have consistently addressed bugs in a timely manner, thanks again and keep up the great work!
